torch/utils/data/dataset.py

```diff
@@ -316,7 +316,10 @@ def __init__(self, dataset: Dataset[T_co], indices: Sequence[int]) -> None:
         self.indices = indices
 
     def __getitem__(self, idx):
-        return self.dataset[self.indices[idx]]
+        if isinstance(self.indices, (list, tuple)) and isinstance(idx, Iterable):
+            return self.dataset[[self.indices[i] for i in idx]]
+        else:
+            return self.dataset[self.indices[idx]]
```

Comment:
Let's just check `idx` is instance of list, because users may not want to iterate over tuple argument.  For other cases, we fall back to `self.dataset[self.indices[idx]]`

nit: you can remove `else` and return directly.

Reply:
Sure makes sense, I will make the changes. Thank you for taking the time to review the code, I appreciate it.